### PR TITLE
Lucide Icon Pack

### PR DESCRIPTION
**Pull request by SnoutBug**
This PR emerged from #4039 and contains only the icons, not the actual implementation of switching icon sets

dark
light

### CODE REVIEW

**Comment by rolandlo** [comment]
Ok, now the icon theme can be chosen from the settings.
iconsCombo

Options are
- Default Light
- Default Dark
- Lucide Light
- Lucide Dark
 
The selection works (after restart).

File: src/core/gui/dialog/SettingsDialog.cpp [hunk 1]
```diff
@@ -301,6 +304,8 @@ void SettingsDialog::customStylusIconTypeChanged() {
     gtk_widget_set_sensitive(get("highlightCursorGrid"), showCursorHighlightOptions);
 }
 
+void SettingsDialog::iconThemeChanged() {}
```
Comment (Technius) [inline src/core/gui/dialog/SettingsDialog.cpp:307]: Is this needed?

File: src/core/control/XournalMain.cpp [hunk 2]
```diff
@@ -408,15 +408,30 @@ void on_startup(GApplication* application, XMPtr app_data) {
...
+        const auto lightLucideIcons = (uiPath / "iconsLucide-light").u8string();
+        const auto darkLucideIcons = (uiPath / "iconsLucide-dark").u8string();
 
         // icon load order from lowest priority to highest priority
-        std::vector<std::string> iconLoadOrder = {lightIcons.u8string()};
-        if (app_data->control->getSettings()->isDarkTheme()) {
-            iconLoadOrder.emplace_back(darkIcons.u8string());
-        } else {
-            iconLoadOrder.insert(iconLoadOrder.begin(), darkIcons.u8string());
+        std::vector<std::string> iconLoadOrder = {};
```
Comment (Technius) [inline src/core/control/XournalMain.cpp:418]: Can you add a comment clarifying that _later_ items in the icon load order have _higher_ priority?

File: src/core/control/settings/SettingsEnums.h [hunk 3]
```diff
@@ -69,6 +69,13 @@ enum StylusCursorType {
     STYLUS_CURSOR_ARROW = 3,
 };
 
+enum IconTheme {
+    ICON_THEME_LIGHT = 0,
+    ICON_THEME_DARK = 1,
+    ICON_THEME_LUCIDE_LIGHT = 2,
+    ICON_THEME_LUCIDE_DARK = 3,
```
Comment (Technius) [inline src/core/control/settings/SettingsEnums.h:76]: I think it's better to have separate options for style and dark/light:
* Keep the "dark theme" option in the settings (this is good for backward compatibility too -- upgrading won't cause users to suddenly lose their dark theme setting).
* Add a "style" option which allows you to select which icon theme you want. As a bonus, this also simplifies icon folder naming/handling, as you just need to define a icon folder name (e.g., iconLucide) and then append the `-dark` or `-light` based on the dark theme setting.

File: src/core/control/XournalMain.cpp (hunk 2 above)
Comment (rolandlo) [reply src/core/control/XournalMain.cpp:418]: There is already a comment, just the line above.

File: src/core/gui/dialog/SettingsDialog.cpp (hunk 1 above)
Comment (rolandlo) [reply src/core/gui/dialog/SettingsDialog.cpp:307]: Currently it is not needed. I will remove it, if I can't find a purpose.

File: src/core/control/settings/SettingsEnums.h (hunk 3 above)
Comment (rolandlo) [reply src/core/control/settings/SettingsEnums.h:76]: Ok, that makes sense in view of the current behavior shown by Xournal++.

**Comment by rolandlo** [comment]
That's how the selection looks now:
iconsComboAndCheckbox

**Comment by rolandlo** [comment]
> That's how the selection looks now: iconsComboAndCheckbox

Values for "Icon Theme" are currently `Color` (which is default) and `Lucide`.

File: src/core/control/XournalMain.cpp (hunk 2 above)
Comment (Technius) [reply src/core/control/XournalMain.cpp:418]: Somehow I missed that, thanks.

**Review by Technius — APPROVED** [review]

**Comment by rolandlo** [comment]
The highlight color issue mentioned by @SnoutBug  in https://github.com/xournalpp/xournalpp/issues/4039#issuecomment-1140520459 is not addressed by this PR. I think it could be done in another PR. It only affects users with a particular highlight color.
@SnoutBug Can you please check the copyright information from f605e4e43e2ae8a3705e4e3798bdf34eaf4f9aff?

Commits 759524916a6296aab44018e2f00018f589c6bbd9, 2801e40993ed8ece8edcf7049adc2dc2905e1045, 98d3c6878b0fc4f341f3a7dfe87e9c77e9d9466a, 6c22cdd0cd1eeb94eb98b3e1e743cc4659d112dc
should be squashed into one commit before merging. I think I can't do it because of force push protection of the branch. @SnoutBug If you can either remove force push protection of the branch or squash the commits yourself (using interactive rebasing, `git rebase -i HEAD~5`, changing the order of the commits so that the commit related to `copyright.txt` is at the end, and marking the later 4 commits as fixups, plus amending the commit description of the squashed commit), please do it. Otherwise I will need to create a new PR with the same content to do it myself.

**Comment by SnoutBug** [comment]
> @SnoutBug If you can either remove force push protection of the branch

Should be removed now, not sure though since I've never worked in this environment this extensively

**Comment by rolandlo** [comment]
Merging in 48 hours if no objections are raised.

**Comment by rolandlo** [comment]
Thanks a lot @SnoutBug for contributing the Lucide Icon Theme! Just merged it.

**Comment by daholzfeind** [comment]
I just checked the latest AppImage of Xournal++, it seems, that Lucide is now the default icon pack for the AppImage variant. 
image

On the other hand, latest deb package is using - correctly - the default Xournal++ icon pack.

**Comment by SnoutBug** [comment]
Also light and dark themes for these icons seem to be mixed up as well for the appimage

**Comment by rolandlo** [comment]
On the AppImage the icon priority order is reversed for some reason. 
https://github.com/xournalpp/xournalpp/blob/9ae2429bffd9906ee3e132242b3a35df3ff7b7c4/src/core/control/XournalMain.cpp#L435-L455
Probably the issue will get resolved automatically once we drop Ubuntu 18.04 support and build the AppImage on Ubuntu 20.04.

**Comment by daholzfeind** [comment]
Hi @rolandlo 

> On the AppImage the icon priority order is reversed for some reason.

The issue is still present in the nightly AppImage version.

**Comment by rolandlo** [comment]
@daholzfeind 
Yes, I know, we haven't fixed that. The AppImage will be built on Ubuntu 20.04 after the 1.2.0 release and that should fix the issue.

**Comment by daholzfeind** [comment]
@rolandlo Thanks for the update!

**Pull request merged**